workspaces/src/worker/mod.rs

```diff
@@ -20,41 +20,41 @@ where
 }
 
 /// Spin up a new sandbox instance, and grab a [`Worker`] that interacts with it.
-pub fn sandbox() -> Worker<Sandbox> {
-    Worker::new(Sandbox::new())
+pub async fn sandbox() -> anyhow::Result<Worker<Sandbox>> {
+    Ok(Worker::new(Sandbox::new().await?))
 }
 
 /// Connect to the [testnet](https://explorer.testnet.near.org/) network, and grab
 /// a [`Worker`] that can interact with it.
-pub fn testnet() -> Worker<Testnet> {
+pub async fn testnet() -> Worker<Testnet> {
     Worker::new(Testnet::new())
 }
 
 /// Connect to the [testnet archival](https://near-nodes.io/intro/node-types#archival-node)
 /// network, and grab a [`Worker`] that can interact with it.
-pub fn testnet_archival() -> Worker<Testnet> {
+pub async fn testnet_archival() -> Worker<Testnet> {
     Worker::new(Testnet::archival())
 }
 
 /// Connect to the [mainnet](https://explorer.near.org/) network, and grab
 /// a [`Worker`] that can interact with it.
-pub fn mainnet() -> Worker<Mainnet> {
+pub async fn mainnet() -> Worker<Mainnet> {
     Worker::new(Mainnet::new())
 }
 
 /// Connect to the [mainnet archival](https://near-nodes.io/intro/node-types#archival-node)
 /// network, and grab a [`Worker`] that can interact with it.
-pub fn mainnet_archival() -> Worker<Mainnet> {
+pub async fn mainnet_archival() -> Worker<Mainnet> {
```

Comment:
are these all made async for a specific reason? Also, why not do a status call on other networks other than sandbox and return a result for the other networks? Feels weird to have this inconsistency

Reply:
yeah I was thinking about consistency, but the Result one was because the other networks didn't return result, but lets go with this:
> feels like you should also make them return results and check the RPC status if so.

normally, RPC for these should never be down, but then again that assumption would be wrong if it ever goes down. I guess by returning a Result, the experience would be better in that particular case

Reply:
Yeah, I think having the UX better in that failing case, or simply having some issue connecting with the RPC for any other reason is enough to keep this in line. Might be confusing for developers also to have different signatures for these.

Opinionated comment, so feel free to disregard, your call

Reply:
went with it since it makes the whole experience much better. Thanks for the suggestion!